app/gui2/public/visualizations/GeoMapVisualization.vue

```diff
@@ -0,0 +1,430 @@
+<script lang="ts">
+export const name = 'Geo Map'
+export const inputType = 'Standard.Table.Data.Table.Table'
+export const scripts = [
+  // mapbox-gl does not have an ESM release.
+  'https://api.tiles.mapbox.com/mapbox-gl-js/v2.15.0/mapbox-gl.js',
+  // The deck.gl scripting API is not available in the ESM module.
+  'https://cdn.jsdelivr.net/npm/deck.gl@8.9.27/dist.min.js',
+]
+
+/**
+ * Provides a mapbox & deck.gl-based map visualization.
+ *
+ * > Example creates a map with described properties with a scatter plot overlay:
+ * {
+ * "latitude": 37.8,
+ * "longitude": -122.45,
+ * "zoom": 15,
+ * "controller": true,
+ * "showingLabels": true, // Enables presenting labels when hovering over a point.
+ * "layers": [{
+ *     "type": "Scatterplot_Layer",
+ *     "data": [{
+ *         "latitude": 37.8,
+ *         "longitude": -122.45,
+ *         "color": [255, 0, 0],
+ *         "radius": 100,
+ *         "label": "an example label"
+ *     }]
+ * }]
+ * }
+ *
+ * Can also consume a dataframe that has the columns `latitude`, `longitude` and optionally `label`.
+ *
+ * TODO: Make 2-finger panning behave like in IDE, and RMB zooming. [#1368]
+ */
+
+type Data = RegularData | Layer | DataFrame
+
+interface RegularData {
+  latitude?: number
+  longitude?: number
+  zoom?: number
+  mapStyle?: string
+  pitch?: number
+  controller?: boolean
+  showingLabels?: boolean
+  layers: Layer[]
+}
+
+interface Layer {
+  type: string
+  data: Location[]
+}
+
+type Color = [red: number, green: number, blue: number]
+
+interface Location {
+  latitude: number
+  longitude: number
+  color?: Color | undefined
+  radius?: number | undefined
+  label?: string | undefined
+}
+
+interface LocationWithPosition {
+  position: [longitude: number, latitude: number]
+  color?: Color
+  radius?: number
+  label?: string
+}
+
+interface DataFrame {
+  df_latitude: number[]
+  df_longitude: number[]
+  df_color?: Color[]
+  df_radius?: number[]
+  df_label?: string[]
+}
+declare var deck: typeof import('deck.gl')
+</script>
+
+<script setup lang="ts">
+/// <reference types="@danmarshall/deckgl-typings" />
+import FindIcon from './icons/find.svg'
+import Path2Icon from './icons/path2.svg'
+import GeoMapDistanceIcon from './icons/geo_map_distance.svg'
+import GeoMapPinIcon from './icons/geo_map_pin.svg'
```

Comment:
should be simple enough to do this for visualizations in `src`, not sure if it's worth doing for those in `/public`, since i think we'd have to duplicate `SvgIcon`

Reply:
You don't have to duplicate anything. You would use it as ```<img :src="`${icons}#geo_map_pin`" />```